backends/dpdk/dpdkAsmOpt.h

```diff
@@ -329,21 +329,32 @@ class CollectUseDefInfo : public Inspector {
 
     bool preorder(const IR::DpdkLearnStatement *b) override {
         usesInfo[b->timeout->toString()]++;
-        if (b->argument)
+        dontEliminate[b->timeout->toString()] = true;
+        if (b->argument) {
             usesInfo[b->argument->toString()]++;
+            // dpdk expect all action argument to be contiguous starting from first argument
+            // passed to learner action
+            dontEliminate[b->argument->toString()] = true;
+        }
         return false;
     }
 
     bool preorder(const IR::DpdkUnaryStatement *u) override {
         usesInfo[u->src->toString()]++;
         defInfo[u->dst->toString()]++;
+        // do not eliminate the destination
+        dontEliminate[u->dst->toString()] = true;
         return false;
     }
 
     bool preorder(const IR::DpdkBinaryStatement *b) override {
         usesInfo[b->src1->toString()]++;
         usesInfo[b->src2->toString()]++;
         defInfo[b->dst->toString()]++;
+        // dst and src1 can not be eliminated, because both are same
+        // and dpdk does not allow src1 to be constant
+        dontEliminate[b->dst->toString()] = true;
```

Comment:
it looks to me like this could be a helper function which takes an expression.
that could be useful particularly if later you decide to change the way you represent the "dontEliminate" information.